extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -581,13 +580,13 @@ private void authorizeTable(
 
   private void authorize(String resource, String key, Action action, HttpServletRequest request)
   {
-    final Access authResult = authorizeAccess(resource, key, action, request);
-    if (!authResult.isAllowed()) {
-      throw new ForbiddenException(authResult.toString());
-    }
+    final AuthorizationResult authResult = authorizeAccess(resource, key, action, request);
+    authResult.getPermissionErrorMessage(true).ifPresent(error -> {
```

Comment:
its weird that we throw a 401 if there is a permission error message. There should be a function available in AuthorizationResult to directly return a boolean indicating whether user is authorized or not.